### PR TITLE
osutil: honor SNAPD_UNSAFE_IO for testing

### PR DESCRIPTION
**Pull request by zyga**
When SNAPD_UNSAFE_IO is set and we are testing (the executable ends with
.test) then skip some sync calls that are very very heavy. Doing this
can easily cut the timy by an order of magnitude.

As a non-scientific but useful measurement, overlord tests go down from
2 minutes 41 seconds to just 15 seconds. This is not representative to
all modules but when it matters, it is noticeable.
```
zyga@faroe:~/snapd/overlord> time SNAPD_UNSAFE_IO=0 go test
OK: 41 passed
PASS
ok  	github.com/snapcore/snapd/overlord	156.342s

real	2m41,765s
user	0m12,964s
sys	0m7,061s

zyga@faroe:~/snapd/overlord> time SNAPD_UNSAFE_IO=1 go test
OK: 41 passed
PASS
ok  	github.com/snapcore/snapd/overlord	9.963s

real	0m15,380s
user	0m12,217s
sys	0m1,626s
```

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

File: osutil/io.go [hunk 1]
```diff
@@ -91,13 +92,29 @@ func AtomicWriteFileChown(filename string, data []byte, perm os.FileMode, flags
...
+	if shouldReallySync {
+		return dir.Sync()
+	}
+	return nil
+}
+
+// Allow disabling sync for testing. This brings massive improvements on
+// certain filesystems (like btrfs) and very much noticeable improvements in
+// all unit tests in genreal.
+var shouldReallySync bool = true
```
Comment (niemeyer) [inline osutil/io.go]: Let's please rename this to snapdUnsafeIO so the relationship to the variable is clear (we need to invert the tests too). We can also drop the = true/false part as it defaults to false.

In fact, we can probably inline:
```
var snapdUsafeIO = len(os.Args) > 0 && strings.HasSuffix(os.Args[0], ".test") && GetenvBool("SNAPD_UNSAFE_IO")

Comment (zyga) [inline osutil/io.go]: Done

File: osutil/io.go [hunk 2]
```diff
@@ -91,13 +92,23 @@ func AtomicWriteFileChown(filename string, data []byte, perm os.FileMode, flags
...
+	if !snapdUnsafeIO {
+		return dir.Sync()
+	}
+	return nil
 }
+
+// Allow disabling sync for testing. This brings massive improvements on
+// certain filesystems (like btrfs) and very much noticeable improvements in
+// all unit tests in genreal.
+var snapdUnsafeIO bool = len(os.Args) > 0 && strings.HasSuffix(os.Args[0], ".test") && GetenvBool("SNAPD_UNSAFE_IO")
```
Comment (mvo5) [inline osutil/io.go]: (nitpick) - I would declare this at the top of the file somewhere

Comment (zyga) [inline osutil/io.go]: Done

**Pull request merged**